src/libraries/Hooks.sol

```diff
@@ -0,0 +1,369 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {ILBHooks} from "../interfaces/ILBHooks.sol";
+
+/**
+ * @title Hooks library
+ * @notice This library contains functions that should be used to interact with hooks
+ */
+library Hooks {
+    error Hooks__CallFailed();
+
+    bytes32 internal constant BEFORE_SWAP_FLAG = bytes32(uint256(1 << 160));
+    bytes32 internal constant AFTER_SWAP_FLAG = bytes32(uint256(1 << 161));
+    bytes32 internal constant BEFORE_FLASH_LOAN_FLAG = bytes32(uint256(1 << 162));
+    bytes32 internal constant AFTER_FLASH_LOAN_FLAG = bytes32(uint256(1 << 163));
+    bytes32 internal constant BEFORE_MINT_FLAG = bytes32(uint256(1 << 164));
+    bytes32 internal constant AFTER_MINT_FLAG = bytes32(uint256(1 << 165));
+    bytes32 internal constant BEFORE_BURN_FLAG = bytes32(uint256(1 << 166));
+    bytes32 internal constant AFTER_BURN_FLAG = bytes32(uint256(1 << 167));
+    bytes32 internal constant BEFORE_TRANSFER_FLAG = bytes32(uint256(1 << 168));
+    bytes32 internal constant AFTER_TRANSFER_FLAG = bytes32(uint256(1 << 169));
+
+    struct Parameters {
+        address hooks;
```

Comment:
The hook contract being just called "hooks" was a bit confusing for me at the beginning. Not sure if worth modifying, but maybe it would be clearer if it was called "hooksContractAddress" or something like this

Reply:
I see what you mean, but I don't know what name to give it tbh, `hooksContractAddress` feels a bit too much imo

Reply:
Yeah make sense, I'm fine with letting it as is